src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -319,6 +320,59 @@ void DoRemoveAllFabrics(intptr_t)
 {
     OpenDefaultPairingWindow(ResetAdmins::kYes);
 }
+
+CHIP_ERROR SendNOCResponse(chip::app::Command * commandObj, EmberAfNOCResponseStatus status, uint8_t index, ByteSpan debug_text)
+{
+    app::CommandPathParams cmdParams = { emberAfCurrentEndpoint(), /* group id */ 0, ZCL_OPERATIONAL_CREDENTIALS_CLUSTER_ID,
+                                         ZCL_NOC_RESPONSE_COMMAND_ID, (chip::app::CommandPathFlags::kEndpointIdValid) };
+    TLV::TLVWriter * writer          = nullptr;
+
+    VerifyOrReturnError(commandObj != nullptr, CHIP_ERROR_INCORRECT_STATE);
+
+    ReturnErrorOnFailure(commandObj->PrepareCommand(cmdParams));
+    writer = commandObj->GetCommandDataElementTLVWriter();
+    ReturnErrorOnFailure(writer->Put(TLV::ContextTag(0), status));
```

Comment:
Can we not use hardcoded constants, but generated field ids instead?

Reply:
Are the field IDs generated in any header files? I did a search through our code base, but couldn't find anything. Even the generated `IMClusterCommandHandler` seems to be using hardcoded constants for the fields.

Reply:
They are not thus generated anywhere.  One of the things that are in very bad shape with our current command handling APIs.  :(

Reply:
Sadness...